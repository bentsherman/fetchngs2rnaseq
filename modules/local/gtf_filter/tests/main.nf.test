nextflow_process {

    name "Test Process GTF_FILTER"
    script "../main.nf"
    process "GTF_FILTER"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
