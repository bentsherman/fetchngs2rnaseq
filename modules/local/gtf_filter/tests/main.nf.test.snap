{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "genome.filtered.gtf:md5,aa8b2aa1e0b5fbbba3b04d471e1b0535"
                ],
                "1": [
                    "versions.yml:md5,4adf55ec05d247fd6d253459bd80856f"
                ],
                "genome_gtf": [
                    "genome.filtered.gtf:md5,aa8b2aa1e0b5fbbba3b04d471e1b0535"
                ],
                "versions": [
                    "versions.yml:md5,4adf55ec05d247fd6d253459bd80856f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-06T15:52:37.977802"
    }
}